### ISSUE TITLE
Padding issue with gridline registered datasets

### ISSUE THREAD

**Issue by meghanrjones** [GenericMappingTools/gmt#6218]
For the gridline registered grids, there is often empty space between the map border and grid edge:
```
gmt grdimage @earth_relief_01d_g -R110/140/20/35 -JB125/20/25/45/12c -B -png test_relief_g
```
![test_synbath_g](https://user-images.githubusercontent.com/14077947/149202966-aa0de428-d21b-4daf-b15d-c561ab375c63.png)

_Originally posted by @meghanrjones in https://github.com/GenericMappingTools/remote-datasets/issues/9#issuecomment-1011348913_

**Comment by PaulWessel** [comment]
The real reason seems to be a combination of this section in grdimage's parse function:

```
	if (!GMT->common.n.active && (!Ctrl->C.active || gmt_is_cpt_master (GMT, Ctrl->C.file)))
		/* Unless user selected -n we want the default not to exceed data range on projection when we are auto-scaling a master table */
		n_errors += gmtinit_parse_n_option (GMT, "b+c");
```

and the changes in gmt_grdio.c. We clearly need the extra row/column and perhaps I can trick the min/max calculation by manipulating the pad temporarily.

**Comment by meghanrjones** [comment]
Fixed by #6272
